internal/prober/multihttp/script_test.go

```diff
@@ -10,52 +10,108 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestBuildUrl(t *testing.T) {
+func TestBuildQueryParams(t *testing.T) {
 	testcases := map[string]struct {
 		request  sm.MultiHttpEntryRequest
 		expected string
 	}{
 		"trivial": {
 			request: sm.MultiHttpEntryRequest{
-				Url: "https://www.example.org/",
+				QueryFields: []*sm.QueryField{
+					{
+						Name:  "q",
+						Value: "hello",
+					},
+				},
 			},
-			expected: "https://www.example.org/",
+			expected: "url.searchParams.append(`q`, `hello`)\n",
 		},
-		"with query fields": {
+		"multiple": {
 			request: sm.MultiHttpEntryRequest{
-				Url: "https://www.example.org/",
 				QueryFields: []*sm.QueryField{
 					{
 						Name:  "q",
 						Value: "hello",
 					},
+					{
+						Name:  "w",
+						Value: "goodbye",
+					},
 				},
 			},
-			expected: "https://www.example.org/?q\\u003Dhello",
+			expected: "url.searchParams.append(`q`, `hello`)\n\turl.searchParams.append(`w`, `goodbye`)\n",
 		},
-		"query without value": {
+		"without value": {
 			request: sm.MultiHttpEntryRequest{
-				Url: "https://www.example.org/",
 				QueryFields: []*sm.QueryField{
 					{
 						Name:  "q",
 						Value: "",
 					},
 				},
 			},
-			expected: "https://www.example.org/?q",
+			expected: "url.searchParams.append(`q`, ``)\n",
+		},
+		"variable in query value": {
+			request: sm.MultiHttpEntryRequest{
+				QueryFields: []*sm.QueryField{
+					{
+						Name:  "q",
+						Value: "${variable}",
+					},
+				},
+			},
+			expected: "url.searchParams.append(`q`, `${vars['variable']}`)\n",
+		},
+		"multiple variables in query value": {
+			request: sm.MultiHttpEntryRequest{
+				QueryFields: []*sm.QueryField{
+					{
+						Name:  "q",
+						Value: "${variable1}and${variable2}",
+					},
+				},
+			},
+			expected: "url.searchParams.append(`q`, `${vars['variable1']}and${vars['variable2']}`)\n",
 		},
-		"query needs encoding": {
+	}
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			actual := buildQueryParams(&tc.request)
+			require.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestBuildUrl(t *testing.T) {
+	testcases := map[string]struct {
+		request  sm.MultiHttpEntryRequest
+		expected string
+	}{
+		"trivial": {
 			request: sm.MultiHttpEntryRequest{
 				Url: "https://www.example.org/",
+			},
+			expected: "new URL(`https://www.example.org/`)",
+		},
+		"variable in url": {
+			request: sm.MultiHttpEntryRequest{
+				Url: "${variable}",
 				QueryFields: []*sm.QueryField{
 					{
-						Name:  "p&q",
```

Comment:
The k6 URL class will escape invalid characters, so it doesn't need to be done here anymore